src/App.js

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from 'react';
+import './App.css';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.handleRock = this.handleRock.bind(this);
+    this.handlePaper = this.handlePaper.bind(this);
+    this.handleScissors = this.handleScissors.bind(this);
+    this.computerPlay = this.computerPlay.bind(this);
+    this.game = this.game.bind(this);
+    this.show = this.show.bind(this);
+    this.handleReset = this.handleReset.bind(this);
+    this.winner = this.winner.bind(this);
+    this.decisionHandler = this.decisionHandler.bind(this);
+
+    this.state = {
+      computer: "",
+      human: "",
+      status: "",
+      computerCount: 0,
+      humanCount: 0,
+      totalCount: 0,
+      ties: 0,
+    }
+  }
+
+  computerPlay = () => {
+    const computerSelection = (Math.floor(Math.random() * 3));
+    if (computerSelection === 0) {
+      return "paper"
+    }
+    if (computerSelection === 1) {
+      return "rock"
+    }
+    if (computerSelection === 2) {
+      return "scissors"
+    }
+  }
+  decisionHandler(choice) {
+    const compState = () => {
+      this.setState((prevState) => ({
+        computer,
+        human: choice,
+        computerCount: prevState.computerCount + 1,
+        totalCount: prevState.totalCount + 1
+      }))
+    }
+
+    const humanState = () => {
+      this.setState((prevState) => ({
+        computer,
+        human: choice,
+        humanCount: prevState.humanCount + 1,
+        totalCount: prevState.totalCount + 1
+      }))
+    }
+
+    const computer = this.computerPlay()
+    if (choice === computer) {
+      this.setState((prevState) => ({
+        computer,
+        human: choice,
+        totalCount: prevState.totalCount + 1,
+        ties: prevState.ties + 1
+      }))
+    }
+    if (choice === "scissors" && computer === "rock") {
+      compState();
+    }
+    if (choice === "scissors" && computer === "paper") {
+      humanState();
+    }
+    if (choice === "rock" && computer === "scissors") {
+      humanState();
+    }
+    if (choice === "rock" && computer === "paper") {
+      compState();
+    }
+    if (choice === "paper" && computer === "rock") {
+      humanState();
+    }
+    if (this.human === "paper" && computer === "scissors") {
+      compState();
+    }
+  }
+  handleRock() {
+    this.decisionHandler("rock")
+  }
+
+  handlePaper() {
+    this.decisionHandler("paper")
+  }
+
+  handleScissors() {
+    this.decisionHandler("scissors")
+  }
+
+  game() {
+    if (this.state.human === this.state.computer) {
+      return (<div>No Winner in this game</div>)
+    }
+    if (this.state.human === "scissors" && this.state.computer === "rock") {
```

Comment:
All of this logic is effectively duplicated in the `decisionHandler()` method, can you think of a way of DRYing this out too?